### PR TITLE
Add atomic-counter

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by timkpaine** [comment]
@conda-forge/help-python all set!

**Comment by timkpaine** [comment]
@conda-forge/core pls :cry:

File: recipes/atomic-counter/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  imports:
+    - atomic_counter
+  requires:
+    - pip
+  commands:
+    - pip check
+
+outputs:
+  - name: atomic-counter
+  - name: atomic_counter
```
Comment (isuruf) [inline recipes/atomic-counter/meta.yaml:39]: Why? It's better to use only one.

Comment (timkpaine) [reply recipes/atomic-counter/meta.yaml:39]: I fundamentally disagree :smile: this is how pypi does it and i've applied this logic across >20 conda forge projects

Comment (isuruf) [reply recipes/atomic-counter/meta.yaml:39]: I'm not saying the recipe is wrong. It's necessary when there are two feedstocks with the two names. However, starting with both seems like an anti-pattern.

Comment (timkpaine) [reply recipes/atomic-counter/meta.yaml:39]: oh i have a huge backlog of projects i still need to go and add -, _ compatibility to. I consider this the preferred pattern after staging a bunch of these. Some references:

https://github.com/conda-forge/conda-forge.github.io/issues/1797
https://github.com/conda-forge/importlib_metadata-feedstock/pull/107
https://github.com/conda-forge/jupyter-packaging-feedstock/pull/34
https://github.com/conda-forge/staged-recipes/pull/22712
https://github.com/conda-forge/staged-recipes/pull/19624
https://github.com/conda-forge/staged-recipes/pull/19452
https://github.com/conda-forge/staged-recipes/pull/19001
etc etc

Comment (timkpaine) [reply recipes/atomic-counter/meta.yaml:39]: here is the latest:
https://github.com/conda-forge/arrow_odbc-feedstock/pull/23
https://github.com/conda-forge/duckdb-feedstock/issues/6#issuecomment-1654025349

**Comment by timkpaine** [comment]
thanks @ocefpaf!

**Pull request merged**